src/item.cpp

```diff
@@ -5625,6 +5625,57 @@ bool item::is_power_armor() const
     return t->power_armor;
 }
 
+int item::get_avg_encumber( const Character &p, encumber_flags flags ) const
+{
+    const islot_armor *t = find_armor_data();
+    if( t == nullptr ) {
+        // handle wearable guns (e.g. shoulder strap) as special case
+        return is_gun() ? volume() / 750_ml : 0;
+    }
+
+    int encumber = 0;
+
+    // Additional encumbrance from non-rigid pockets
+    float relative_encumbrance = 1;
+    if( !( flags & encumber_flags::assume_full ) ) {
+        relative_encumbrance = contents.relative_encumbrance();
+    }
+
+    for( const armor_portion_data &data : t->data ) {
+        if( data.covers.has_value() ) {
+            encumber += data.encumber;
+            encumber += std::ceil( relative_encumbrance * ( data.max_encumber - data.encumber ) );
+        }
+    }
+    encumber /= t->data.size();
+
+    // Fit checked before changes, fitting shouldn't reduce penalties from patching.
+    if( has_flag( flag_FIT ) && has_flag( flag_VARSIZE ) ) {
+        encumber = std::max( encumber / 2, encumber - 10 );
+    }
+
+    // TODO: Should probably have sizing affect coverage
+    const sizing sizing_level = get_sizing( p );
+    switch( sizing_level ) {
+        case sizing::small_sized_human_char:
+        case sizing::small_sized_big_char:
+            // non small characters have a HARD time wearing undersized clothing
+            encumber *= 3;
+            break;
+        case sizing::human_sized_small_char:
+        case sizing::big_sized_small_char:
+            // clothes bag up around smol characters and encumber them more
+            encumber *= 2;
+            break;
+        default:
+            break;
+    }
+
+    encumber += static_cast<int>( std::ceil( get_clothing_mod_val( clothing_mod_type_encumbrance ) ) );
+
+    return encumber;
+}
```

Comment:
Now I see the code duplication that arises from separating this function I can see why you liked the "default argument" approach to this calculation.

However, we certainly want to avoid this much code duplication.  I can offer a few options to resolve it:
* In `get_avg_encumber` just call `get_encumber` for each bodypart and average.  I think this is the simplest solution.  It's not as performant, but I would guess that none of the places calculating average encumbrance is performance-critical.
* Write a private helper function that uses the default argument or an analogous trick and use it to implement each of `get_avg_encumber` and `get_encumber`.
* Factor out the common parts of these two functions into helper functions, but keep the differing loop in each.